kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/config/UnresolvedKeyPolicy.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption.config;
+
+/**
+ * This policy determines what the Filter should do with records when we cannot
```

Comment:
should it be a record policy, or a ProduceRequest policy?

Reply:
My instinct is for it to be a record policy, so that we apply graceful degradation and minimise blast radius from missing encryption keys, but I have my doubts about how well that actually works in practice as we start sending different requests on the upstream side to the downstream side.